src/rgw/kafka_compat.cc

```diff
@@ -0,0 +1,41 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include "kafka_compat.h"
+#include <errno.h>
+
+#if RD_KAFKA_VERSION < 0x00092ff
+rd_kafka_resp_err_t rd_kafka_flush (rd_kafka_t *rk, int timeout_ms) {
+        unsigned int msg_cnt = 0;
+	int qlen;
+	rd_ts_t ts_end = rd_timeout_init(timeout_ms);
+        int tmout;
+
+	if (rk->rk_type != RD_KAFKA_PRODUCER)
+		return RD_KAFKA_RESP_ERR__NOT_IMPLEMENTED;
+
+        rd_kafka_yield_thread = 0;
+
+        /* First poll call is non-blocking for the case
+         * where timeout_ms==RD_POLL_NOWAIT to make sure poll is
+         * called at least once. */
+        tmout = RD_POLL_NOWAIT;
+        do {
+                rd_kafka_poll(rk, tmout);
```

Comment:
the indent does not look right.

Reply:
> the indent does not look right.

it is copy&paste from the original code. will fix.

Reply:
@tchaikov, ideally this is not needed at all. in shaman (and my local ubuntu18.04 setup), when installing librdkafka on ubuntu you get librdkafka0.11.3, but in jenkins there is librdkafka0.8.6 - which is causing compatibility issues

Reply:
nevermind, building ok now, with the compatibility commit